design/types.go

```diff
@@ -133,6 +133,8 @@ const (
 	UserTypeKind
 	// MediaTypeKind represents a media type.
 	MediaTypeKind
+	// NoneKind represents a nil
```

Comment:
Is that really needed? adding a new type has a lot of consequences... all generators will have to be updated.